### DISCUSSION TITLE
Camply Feedback

### DISCUSSION THREAD

**Discussion by juftin** [juftin/camply#12] [General]
`camply` ⛺️ is finally public! I hope everyone enjoys this project and it actually helps you to get to the places you love. I want to know what you think. Any feedback, bug reports, or feature requests are welcome.

**Comment by juftin** [comment]
All New Tasks, Bugs, Etc will be tracked here as well: https://github.com/juftin/camply/projects/1

**Comment by grantland** [comment]
This looks like an awesome project! Some feedback/ideas I have:

* Some campsites are showing up as "reservable", but on reservation.gov it shows up as "Checkout Only". I added some additional logs and it seems like the API returns them as "Open" though, so I'm not sure anything can be done here
* I want to be notified of any openings within a date range so I used `--continuous --notifications email -- notify-first-try`, but it seems like the loop exits if any reservations are found within the first loop. Am I doing something wrong here or might be understanding the docs wrong?
* I don't seem to get an email using `--continuous --notifications email -- notify-first-try` if there's a reservation found in the first loop.
* It would be great if Pushbullet support could be added!
* Some sort of `config.yml` with campsites/datetime/etc + daemon support would be great so I don't have to keep track of manual instances

Thanks for publishing this project btw

**Reply by juftin** [reply]
@grantland :

> * Some campsites are showing up as "reservable", but on reservation.gov it shows up as "Checkout Only". I added some additional logs and it seems like the API returns them as "Open" though, so I'm not sure anything can be done here

Do you have an examples for this? A URL or campground / recreation area ID + some dates.  I'm guessing I just need to add the `Checkout Only` availability status to this exclusion config  [here](https://github.com/juftin/camply/blob/a5f5b09a32783bfa4b270d15f2d984ff92152d3a/camply/config/api_config.py#L100-L107). 

> * I want to be notified of any openings within a date range so I used `--continuous --notifications email -- notify-first-try`, but it seems like the loop exits if any reservations are found within the first loop. Am I doing something wrong here or might be understanding the docs wrong?

The loop will exit after the search is found if you don't pass the `--search-forever` argument. That will continue running - but never notify twice about the same campsite for the same night.

> * I don't seem to get an email using `--continuous --notifications email -- notify-first-try` if there's a reservation found in the first loop.

hmm... this one seems a little strange. You should still get an email for up to 5 campsites if they were found on the first try. I will try to replicate on my end. 

> * It would be great if Pushbullet support could be added!

Pushbullet is a good idea, I think it should be really easy to integrate with. I'll take a look and wire something up.

> * Some sort of `config.yml` with campsites/datetime/etc + daemon support would be great so I don't have to keep track of manual instances

Yess!!! I was deciding whether or not to build a YAML integration, I will definitely add this to the backlog!

> Thanks for publishing this project btw

You're very welcome! Thank you very much for this feedback, it's greatly appreciated.

**Reply by grantland** [reply]
Thanks for the quick reply!

> Do you have an examples for this? A URL or campground / recreation area ID + some dates. I'm guessing I just need to add the `Checkout Only` availability status to this exclusion config [here](https://github.com/juftin/camply/blob/a5f5b09a32783bfa4b270d15f2d984ff92152d3a/camply/config/api_config.py#L100-L107).

Yeah, here's an example where Nov 27 through Dec 7 is "Checkout Only": https://www.recreation.gov/camping/campsites/61117

<img width="606" alt="Screen Shot 2021-05-24 at 9 04 54 PM" src="https://user-images.githubusercontent.com/277030/119437792-b2decf80-bcd3-11eb-81f7-5c5fa4ad8b12.png">

```
2021-05-25 03:58:57,923 [    INFO]: 📅 Sat, November 27 🏕  1 sites
2021-05-25 03:58:57,924 [    INFO]: 	⛰️  Tahoe National Forest, CA  🏕  Calpine Lookout: ⛺ 1 sites
2021-05-25 03:58:57,925 [    INFO]: 		🔗 https://www.recreation.gov/camping/campsites/61117
2021-05-25 03:58:57,925 [    INFO]: 📅 Sun, November 28 🏕  1 sites
2021-05-25 03:58:57,926 [    INFO]: 	⛰️  Tahoe National Forest, CA  🏕  Calpine Lookout: ⛺ 1 sites
2021-05-25 03:58:57,926 [    INFO]: 		🔗 https://www.recreation.gov/camping/campsites/61117
```

> The loop will exit after the search is found if you don't pass the `--search-forever` argument. That will continue running - but never notify twice about the same campsite for the same night.

Derp, `--search-forever` works!

> hmm... this one seems a little strange. You should still get an email for up to 5 campsites if they were found on the first try. I will try to replicate on my end.

So this line seems to be swallowing errors and should probably be removed:

https://github.com/juftin/camply/blob/a5f5b09a32783bfa4b270d15f2d984ff92152d3a/camply/utils/camply_cli.py#L45

Once removed, I discovered I had some issues with my STMP configuration which I fixed and started getting emails

> Pushbullet is a good idea, I think it should be really easy to integrate with. I'll take a look and wire something up.

> Yess!!! I was deciding whether or not to build a YAML integration, I will definitely add this to the backlog!

These would be amazing, will look forward to them!

**Reply by juftin** [reply]
Thanks again for the great feedback @grantland . The first fix is out! https://github.com/juftin/camply/releases/tag/v0.1.2. A simple `pip install --upgrade camply` is all you need to run.

This update addresses the `Checkout-Only` scenario, fixes the error trap, and performs email connection validation before the search begins. I'll follow up when there is more news on the feature requests.

... also, that [Fire Tower Lookout](https://www.recreation.gov/camping/campgrounds/234131) looks sweet! let me know if you're able to snag a spot

**Reply by grantland** [reply]
Thanks for the quick update, all those problems seem to be fixed! Will let you know if I end up getting a Fire Lookout Tower and if I do it'll be thanks to you!

I've run into another small issue with `--search-forever` when it comes across a campsite I've already been notified about. It displays that a reservable campsite has been found, 0 new campsites found, but still sends an email, albeit blank:

```
2021-05-25 17:37:14,985 [    INFO]: ⛺️ ⛺️ ⛺️ ⛺️ 1 Reservable Campsites Matching Search Preferences
2021-05-25 17:37:14,998 [    INFO]: 0 New Campsites Found.
2021-05-25 17:37:15,984 [    INFO]: Sending Email to <email@address.com>: Camply Notification
2021-05-25 17:37:16,706 [    INFO]: Email sent successfully
```

**Reply by juftin** [reply]
Yikes, you're right 🤦‍♂️. Update coming soon

**Comment by juftin** [comment]
## [Camply 0.1.4](https://github.com/juftin/camply/compare/v0.1.3...v0.1.4) was just released!

### Added

- Pushbullet Notifications
- YAML Search Configuration Files

### Fixed

- PyPi Publishing CI on Release

---------
---------

Thanks to @grantland for some really great feature ideas

**Comment by svetlunamakarunya** [comment]
Hello,

I was trying to use Camply to monitor the wilderness permits for Yosemite, but it seems that it's not yet possible. Would you be so kind to make this available?
Here is the link to what i mean - https://www.recreation.gov/permits/445859/registration/detailed-availability?type=overnight-permit&date=2022-04-29
Or "Explore available permits" from here - https://www.recreation.gov/permits/445859 


Thank you!!!

**Reply by juftin** [reply]
Hey @svetlunamakarunya there is some discussion around this over in Discussions: https://github.com/juftin/camply/discussions/22

Unfortunately the Permits API Endpoints aren't standardized and this is a pretty hefty feature implementation. Any contributions are welcome

**Comment by sanchezluciano** [comment]
I LOVE this project. I was able to get a spot in Yosemite within 24 hours of running this. Thank you so much for making this and taking time to make improvements. Happy camping!
